src/frontend/packages/core/src/core/utils.service.ts

```diff
@@ -280,3 +280,7 @@ export const sortStringify = (obj: { [key: string]: string }): string => {
     return res += `${key}-${obj[key]},`;
   }, '');
 };
+
+export function cloneObject(obj: any) {
```

Comment:
fixed

Reply:
@KlapTrap

Reply:
I've looked at all the places we use deepClone and in every case I think it can be replaced with something simpler. As an example, doing a deep clone on PolicyDefaultTrigger is no different than doing the simpler {...PolicyDefaultTrigger} as all the values are primitive - in that example I would go even further and change PolicyDefaultTrigger into a function that just returns a new object with the default values each time. Doing this completely removing the need to clone anything.

Reply:
I'm also wondering if we should get rid of `initIfUndefined` too. It seems like we can reduce the complexity by simply doing `newPolicy.scaling_rules = newPolicy.scaling_rules || []` rather than `newPolicy = initIfUndefined(newPolicy, 'scaling_rules', []) as AppAutoscalerPolicyLocal;` (as an example).

Could we change `initIfUndefined` to do something like `newPolicy.scaling_rules = newPolicy.scaling_rules || []` rather than cloning the object every time?

Reply:
Good idea~ fixed